### PR TITLE
fix drop table

### PR DESCRIPTION
**Pull request by siddontang**
fix https://github.com/pingcap/tidb/issues/457

### CODE REVIEW

File: structure/hash.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func (t *TxStructure) HLen(key []byte) (int64, error) {
 func (t *TxStructure) HDel(key []byte, fields ...[]byte) error {
 	metaKey := t.encodeHashMetaKey(key)
 	meta, err := t.loadHashMeta(metaKey)
-	if err != nil {
+	if err != nil || meta.IsEmpty() {
 		return errors.Trace(err)
```
Comment (coocood) [inline structure/hash.go]: line 170 `if meta.IsEmpty() {` become dead code after this change.

Comment (siddontang) [inline structure/hash.go]: no, the hash can become empty after deleting some fields too.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**